### PR TITLE
fix: multiple smaller problems in generic sync

### PR DESCRIPTION
**Pull request by FabianKramm**
**What issue type does this pull request address?** (keep at least one, remove the others) 
/kind bugfix

**Please provide a short message that should be published in the vcluster release notes**
Fixed an issue where vcluster would not correctly sync tekton resources

### CODE REVIEW

File: pkg/controllers/generic/export_syncer.go [hunk 1]
```diff
@@ -46,7 +46,17 @@ func CreateExporters(ctx *context.ControllerContext, config *config.Config) erro
...
+	for _, exportConfig := range exporterConfig.Exports {
+		reversePatches := []*config.Patch{
+			{
+				Operation: config.PatchTypeCopyFromObject,
+				FromPath:  "status",
+				Path:      "status",
+			},
+		}
+		reversePatches = append(reversePatches, exportConfig.ReversePatches...)
+		exportConfig.ReversePatches = reversePatches
```
Comment (matskiv) [inline pkg/controllers/generic/export_syncer.go:58]: What are the advantages or use cases of the implicit status reverse patch? Besides shorter config of course. :)

File: cmd/vclusterctl/cmd/delete.go [hunk 2]
```diff
@@ -153,6 +154,28 @@ func (cmd *DeleteCmd) Run(cobraCmd *cobra.Command, args []string) error {
 		} else {
 			cmd.log.Donef("Successfully deleted virtual cluster namespace %s", cmd.Namespace)
 		}
+
+		// delete multi namespace mode namespaces
```
Comment (matskiv) [inline cmd/vclusterctl/cmd/delete.go:158]: Should we mention this new functionality in the description of the `--delete-namespace` flag?

**Review by matskiv — COMMENTED** [review]
Overall it looks good, but I have a few questions before approving.

File: pkg/controllers/generic/import_syncer.go [hunk 3]
```diff
@@ -107,6 +107,43 @@ func (s *importer) Name() string {
...
+		return true
+	}
+	if obj.GetAnnotations() != nil &&
+		obj.GetAnnotations()[translate.ControllerLabel] != "" &&
+		obj.GetAnnotations()[translate.ControllerLabel] != s.Name() {
+		// make sure kind matches
+		splitted := strings.Split(obj.GetAnnotations()[translate.ControllerLabel], "/")
+		if len(splitted) != 3 {
+			return true
+		} else if splitted[0] != strings.ToLower(obj.GetObjectKind().GroupVersionKind().Kind) {
```
[the remark this answers is not in the document]
Comment (FabianKramm) [reply pkg/controllers/generic/import_syncer.go:138]: Good catch, we should check that as well

File: pkg/util/translate/translate.go [hunk 4]
```diff
@@ -241,6 +241,21 @@ func EnsureCRDFromPhysicalCluster(ctx context.Context, pConfig *rest.Config, vCo
...
+	newVersions := []apiextensionsv1.CustomResourceDefinitionVersion{}
+	for _, version := range crdDefinition.Spec.Versions {
+		if version.Name == groupVersionKind.Version {
+			version.Served = true
+			version.Storage = true
+			newVersions = append(newVersions, version)
+			break
+		}
+	}
+	crdDefinition.Spec.Versions = newVersions
```
[the remark this answers is not in the document]
Comment (FabianKramm) [reply pkg/util/translate/translate.go:256]: Problem is the conversion webhook here as it points to a service in the host cluster and probably won't work for vcluster crds. Without the conversion webhook, Kubernetes cannot auto upgrade those versions (especially if the version selected is not the one stored), which is why we only can copy 1 to the vcluster.

File: pkg/controllers/generic/export_syncer.go (hunk 1 above)
Comment (FabianKramm) [reply pkg/controllers/generic/export_syncer.go:58]: its mostly shorter config, the question would be why not to include it, as you probably always want it

File: cmd/vclusterctl/cmd/delete.go (hunk 2 above)
Comment (FabianKramm) [reply cmd/vclusterctl/cmd/delete.go:158]: Yeah I'll add a mention!

File: pkg/util/translate/translate.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (matskiv) [reply pkg/util/translate/translate.go:256]: So to support the version upgrade we would need to implement some sort of proxy endpoint in the syncer? :thinking:

[the remark this answers is not in the document]
Comment (matskiv) [reply pkg/util/translate/translate.go:256]: I created a follow up feature request for this - https://github.com/loft-sh/vcluster/issues/909

**Review by matskiv — APPROVED** [review]

**Pull request merged**